packages/vector_graphics_compiler/lib/src/geometry/path.dart

```diff
@@ -0,0 +1,475 @@
+import 'dart:math' as math;
+
+import 'package:meta/meta.dart';
+import 'package:path_parsing/path_parsing.dart';
+
+import 'basic_types.dart';
+import 'matrix.dart';
+import '../util.dart';
+
+const double _kArcApproximationMagic = 0.551915024494;
+
+/// Specifies the winding rule that decies how the interior of a [Path] is
+/// calculated.
+///
+/// This enum is used by the [Path.fillType] property.
+///
+/// It is compatible with the same enum in `dart:ui`.
+enum PathFillType {
+  /// The interior is defined by a non-zero sum of signed edge crossings.
+  ///
+  /// For a given point, the point is considered to be on the inside of the path
+  /// if a line drawn from the point to infinity crosses lines going clockwise
+  /// around the point a different number of times than it crosses lines going
+  /// counter-clockwise around that point.
+  nonZero,
+
+  /// The interior is defined by an odd number of edge crossings.
+  ///
+  /// For a given point, the point is considered to be on the inside of the path
+  /// if a line drawn from the point to infinity crosses an odd number of lines.
+  evenOdd,
+}
+
+/// The available types of path verbs.
+///
+/// Used by [PathCommand.type].
+enum PathCommandType {
+  /// A path verb that picks up the pen to move it to another coordinate,
+  /// starting a new contour.
+  move,
+
+  /// A path verb that draws a line from the current point to a specified
+  /// coordinate.
+  line,
+
+  /// A path verb that draws a Bezier curve from the current point to a
+  /// specified point using two control points.
+  cubic,
+
+  /// A path verb that draws a line from the current point to the starting
+  /// point of the current contour.
+  close,
+}
+
+/// An abstract, immutable representation of a path verb and its associated
+/// points.
+///
+/// [Path] objects are collections of [PathCommand]s. To create a path object,
+/// use a [PathBuilder]. To create a path object from an SVG path definition,
+/// use [parseSvgPathData].
+@immutable
+abstract class PathCommand {
+  const PathCommand._(this.type);
+
+  /// The type of this path command.
+  final PathCommandType type;
+
+  /// Returns a new path command transformed by `matrix`.
+  PathCommand transformed(AffineMatrix matrix);
+}
+
+class LineToCommand extends PathCommand {
+  const LineToCommand(this.x, this.y) : super._(PathCommandType.line);
+
+  /// The absolute offset of the destination point for this path from the x
+  /// axis.
+  final double x;
```

Comment:
Here and elsewhere, since you've defined it consider using the `Point` class for these

Reply:
I'm a bit torn on this - its easier to write the code this way fo rme and feels closer to the dart:ui API it's immitating. I'll leave it for now.